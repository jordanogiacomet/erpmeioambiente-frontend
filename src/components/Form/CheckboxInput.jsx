import React from 'react';
import styled from 'styled-components';
import P from 'prop-types';

const CheckboxWrapper = styled.div`
  display: flex;
  align-items: center;
`;

const CheckboxLabel = styled.label`
  margin-left: 8px;
  color: ${({ theme }) => theme.colors.secondary};
  font-size: ${({ theme }) => theme.font.sizes.large};
  font-weight: 600;
`;

const CheckboxField = styled.input`
  width: auto;
  cursor: pointer;
`;

const CheckboxInput = ({ label, id, ...props }) => (
  <CheckboxWrapper>
    <CheckboxField type="checkbox" id={id} {...props} />
    <CheckboxLabel htmlFor={id}>{label}</CheckboxLabel>
  </CheckboxWrapper>
);

CheckboxInput.propTypes = {
  label: P.string.isRequired,
  id: P.string.isRequired,
};

export default CheckboxInput;
