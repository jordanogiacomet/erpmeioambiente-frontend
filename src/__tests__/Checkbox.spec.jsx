/* eslint-disable testing-library/no-node-access */
import React from 'react';
import renderTheme from '../styles/render-theme';
import { screen } from '@testing-library/react';
import 'jest-styled-components';
import CheckboxInput from '../components/Form/CheckboxInput';

describe('<CheckboxInput />', () => {
  it('should renders correctly with a label', () => {
    renderTheme(<CheckboxInput label="remember me"/>);
    expect(screen.getByText('remember me')).toBeInTheDocument();
  });
  it('should render with input type checkbox', () => {
    renderTheme(<CheckboxInput label="remember me" id="remember-me" />);
    const input = screen.getByRole('checkbox');
    expect(input).toHaveAttribute('type', 'checkbox');
  });
  it('should match snapshot', () => {
    const { container } = renderTheme(<CheckboxInput label="remember me" id="remember me"/>);
    expect(container).toMatchSnapshot();
  })
});