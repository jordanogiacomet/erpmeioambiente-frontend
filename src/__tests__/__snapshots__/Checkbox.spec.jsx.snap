// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CheckboxInput /> should match snapshot 1`] = `
.c0 {
  display: flex;
  align-items: center;
}

.c2 {
  margin-left: 8px;
  color: #FFFFFF;
  font-size: 1.6rem;
  font-weight: 600;
}

.c1 {
  width: auto;
  cursor: pointer;
}

<div>
  <div
    class="c0"
  >
    <input
      class="c1"
      id="remember me"
      type="checkbox"
    />
    <label
      class="c2"
      for="remember me"
    >
      remember me
    </label>
  </div>
</div>
`;
